airbyte-integrations/connectors/source-jina-ai/source_jina_ai/manifest.yaml

```diff
@@ -0,0 +1,158 @@
+          api_token: "{{ config['api_key'] }}"
+        request_headers:
+          Accept: application/json
+          X-With-Links-Summary: "{{ config['gather_links'] }}"
+          X-With-Images-Summary: "{{ config['gather_images'] }}"
+    name: "reader"
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $ref: "#/schemas/reader"
+
+  search_stream:
+    type: DeclarativeStream
+    retriever:
+      type: SimpleRetriever
+      record_selector:
+        $ref: "#/definitions/selector"
+      paginator:
+        type: NoPagination
+      requester:
+        type: HttpRequester
+        url_base: "https://s.jina.ai/{{ config['search_prompt'] }}"
+        http_method: "GET"
+        path: "/"
+        authenticator:
+          type: BearerAuthenticator
+          api_token: "{{ config['api_key'] }}"
+        request_headers:
+          Accept: application/json
+          X-With-Links-Summary: "{{ config['gather_links']}}"
+          X-With-Images-Summary: "{{ config['gather_images']}}"
+    name: "search"
+    $parameters:
+      path: "/example"
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $ref: "#/schemas/search"
+
+streams:
+  - "#/definitions/reader_stream"
+  - "#/definitions/search_stream"
+
+check:
+  type: CheckStream
+  stream_names:
+    - "reader"
+    - "search"
+
+schemas:
+  search:
+    $schema: http://json-schema.org/draft-07/schema#
+    title: Jina AI URL Reader Schema
+    type: object
+    additionalProperties: true
+    properties:
+      title:
+        type: string
+      url:
+        type: string
+      content:
+        type: string
+      description:
+        type: string
+      links:
+        type: object
+        additionalProperties: true
+        properties:
+          More information...:
+            type: string
+
+  reader:
+    $schema: http://json-schema.org/draft-07/schema#
+    title: Jina AI prompt Reader Schema
+    type: object
+    additionalProperties: true
+    properties:
+      title:
+        type: string
+      url:
+        type: string
+      content:
+        type: string
+      description:
+        type: string
+      links:
+        type: object
+        additionalProperties: true
+        properties:
+          More information...:
+            type: string
+
+spec:
+  type: Spec
+  documentation_url: https://docs.airbyte.com/integrations/sources/jina-ai
+  connection_specification:
+    title: Jina Ai Spec
+    type: object
+    required:
+      - api_key
+    additionalProperties: true
+    properties:
+      api_key:
+        type: string
+        title: API Key
+        description: >-
+          Your Jina AI API Key. See <a
+          href="https://jina.ai/reader/#apiform">here</a>.
+        airbyte_secret: true
+        order: 0
+      read_prompt:
+        type: string
+        title: Your read prompt url, that would be appended at the end of reader stream
+        default: "https://www.google.com"
+        order: 1
```

Comment:
Can we accept an array of URLs strings?

And if so, perhaps a rename to match the use case? Maybe something like `reader_urls` or `reader_page_urls`?

If too difficult to implement, let me know. I understand it will change the endpoint and "get" protocol to make this change.

I imagine that the same `reader` stream could produce multiple records in each sync, with the URL serving as primary key for each record.

Reply:
Array of URL's is not supported I guess, the [platform](https://jina.ai/reader/#apiform) example is also taking one at a time.
:thinking:

Reply:
@btkcodedev - I did some digging and I may have found a workaround for this issue. What do you think of this approach?

1. We would use the "parameterized requests" option in the No-Code UI Builder.
2. We'd pass an array input from the user config. (Example in screenshot below.)
3. We'd use the value from the parameterized input for the request itself, instead of passing directly from the config.

![image](https://github.com/airbytehq/PyAirbyte-Hackathon/assets/18150651/fd2f7528-794f-4b4f-8d6b-48896e17aee0)

Totally okay if this doesn't work (I didn't have a chance to test it), but if you are able to give it a shot and report back, that's much appreciated. 🙏

Reply:
@aaronsteers I've tried it but has a catch, the parametrized request is the ListPartitionRouter in lowcode
the injection of that partition can be only done inside body, parameters or headers but not on path or url

Ref: 
![image](https://github.com/airbytehq/airbyte/assets/47392334/d4335495-c77d-4b3a-a4d5-0191e4664875)

So failed on that one

Error Ref: `Jinja macro has undeclared variables: {'stream_partition'}.`
![image](https://github.com/airbytehq/airbyte/assets/47392334/279cea68-96e7-45ca-b0b7-cc49cb821bd6)

Reply:
Sorryy :bow: